### PR TITLE
Updated binary encoding for BlockHeader

### PR DESCRIPTION
**Pull request by fosskers**
See https://github.com/kadena-io/chainweb-node/issues/954

### CODE REVIEW

**Comment by fosskers** [comment]
This needs to be ran on devnet, and possibly also mainnet, to prove that assumptions about hashing aren't being broken.

File: src/Chainweb/BlockHeader.hs [hunk 1]
```diff
@@ -648,28 +648,29 @@ decodeBlockHeaderWithoutHash = do
...
+    ht <- decodeHashTarget
+    pa <- decodeBlockPayloadHash
+    ci <- decodeChainId
+    wt <- decodeBlockWeight
+    hg <- decodeBlockHeight
+    cv <- decodeChainwebVersion
+    es <- decodeEpochStartTime
+    no <- decodeNonce
+    bh <- decodeBlockHash
+    pure $ BlockHeader no ct ph hr ht pa ci wt hg cv es ff bh
```
Comment (larskuhtz) [inline src/Chainweb/BlockHeader.hs]: I think changing the order of the fields in `BlockHeader` wouldn't break anything and would avoid these changes.

Comment (fosskers) [reply src/Chainweb/BlockHeader.hs]: Ah, yes, perhaps that's better.

**Comment by fosskers** [comment]
I'm currently doing a full mainnet catch up on my node with this.

**Pull request merged**